be/src/Unic.UrlMapper2/code/Services/RedirectionService.cs

```diff
@@ -102,8 +102,8 @@ protected virtual Redirect GetRegexMatch(RedirectSearchData redirectSearchData,
             if (string.IsNullOrWhiteSpace(sourceTerm)) return default;
 
             // We are going to take the one regex redirect which has the longest match within the term
-            var regexMatches = enumerableRedirects.Where(r => r.RegexEnabled && Regex.IsMatch(input: sourceTerm, pattern: r.Term));
-            var regexMatch = regexMatches.OrderByDescending(r => r.Term.Length).FirstOrDefault();
+            var regexMatches = enumerableRedirects.Where(r => r.RegexEnabled && !string.IsNullOrWhiteSpace(r.Term) && Regex.IsMatch(input: sourceTerm, pattern: r.Term));
```

Comment:
I've done some additional tests on UrlMapper project itself and I'm almost sure that the problem with empty Source Term could cause the issue also before.